src/main/java/com/esri/samples/localserver/local_server_services/LocalServerServicesController.java

```diff
@@ -0,0 +1,228 @@
+
+  @FXML private Button btnStartServer;
+  @FXML private Button btnStopServer;
+  @FXML private Button btnStartService;
+  @FXML private Button btnStopService;
+  @FXML private Button btnURL;
+  @FXML private ComboBox<String> serviceOptions;
+  @FXML private TextArea statusLog;
+  @FXML private ListView<String> runningServices;
+
+  private LocalServer server;
+  private LocalServer.StatusChangedListener serverStatusListener;
+  private ListenableList<LocalService> services;
+  private HostServices hostServices;
+
+  /** 
+   * Handles starting a local server.
+   * 
+   * @param event the event that fired this action 
+   */
+  @FXML
+  protected void handleStartLocalServer() {
+
+    // start local server
+    server = LocalServer.INSTANCE;
+    serverStatusListener = status -> {
+      statusLog.appendText("Server Status: " + status.getNewStatus().toString() + "\n");
+      if (status.getNewStatus() == LocalServerStatus.STOPPED) {
+        server.removeStatusChangedListener(serverStatusListener);
+      }
+    };
+    server.addStatusChangedListener(serverStatusListener);
+    server.startAsync();
+
+    // tracking services attached to this local server
+    services = server.getServices();
+
+    // update buttons on layout
+    btnStartServer.setDisable(true);
+    btnStopServer.setDisable(false);
+    btnStartService.setDisable(false);
+  }
+
+  /** 
+   * Handles stopping a local server.
+   * 
+   * @param event the event that fired this action 
+   */
+  @FXML
+  protected void handleStopLocalServer() {
+
+    terminate();
+    btnStartServer.setDisable(false);
+    btnStopServer.setDisable(true);
+    btnStartService.setDisable(true);
+    btnStopService.setDisable(true);
+  }
+
+  /**
+   * Gets the selected item from the combo box and creates a Local Service to add to the Local Server.
+   * <p>
+   * Only creates a Local Service that is not currently running on the Local Server.
+   * 
+   * @param event the event that fired this action
+   * @throws URISyntaxException if URL passed to Local Service is incorrect 
+   */
+  @FXML
+  protected void handleStartLocalService(ActionEvent event) throws URISyntaxException {
+
+    if (server.getStatus() == LocalServerStatus.STARTED) {
+      // check that the service selected is not already running
+      boolean serviceExists = false;
+      String serviceSelected = serviceOptions.getSelectionModel().getSelectedItem();
+      for (String runningService : runningServices.getItems()) {
+        if (runningService.startsWith(serviceSelected)) {
+          serviceExists = true;
+        }
+      }
+
+      // start service if not currently running on the Local Server
+      if (!serviceExists) {
+        String serviceUrl = "";
+        LocalService localService = null;
+        switch (serviceSelected) {
+          case "Map Service":
+            serviceUrl = Paths.get(getClass().getResource("/local_server/PointsofInterest.mpk").toURI()).toString();

```

Comment:
I've added them. Delete your samples_data folder and refresh the project to download the new data. Make sure this data is approved for public use.